src/bonsai/rest/leverage.ts

```diff
@@ -0,0 +1,123 @@
+import { DEFAULT_LEVERAGE_PPM } from '@/constants/leverage';
+import { timeUnits } from '@/constants/time';
+
+import { type RootStore } from '@/state/_store';
+import { setSelectedMarketLeverages } from '@/state/raw';
+
+import { parseToPrimitives } from '@/lib/parseToPrimitives';
+import { sleep } from '@/lib/timeUtils';
+import { isPresent } from '@/lib/typeUtils';
+
+import { loadableIdle } from '../lib/loadable';
+import { mapLoadableData } from '../lib/mapLoadable';
+import { createValidatorQueryStoreEffect } from './lib/indexerQueryStoreEffect';
+import { queryResultToLoadable } from './lib/queryResultToLoadable';
+
+export function setUpUserLeverageParamsQuery(store: RootStore) {
+  const cleanupEffect = createValidatorQueryStoreEffect(store, {
+    name: 'leverageParams',
+    selector: () => true,
```

Comment:
this should return your dydxAddress so if you switch accounts it will give a new queryKey